Sources/AsyncHTTPClient/ConnectionPool/State Machine/HTTPConnectionPool+HTTP2StateMachine.swift

```diff
@@ -0,0 +1,116 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the AsyncHTTPClient open source project
+//
+// Copyright (c) 2021 Apple Inc. and the AsyncHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of AsyncHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+import NIOHTTP2
+
+extension HTTPConnectionPool {
+    struct HTTP2StateMaschine {
+        typealias Action = HTTPConnectionPool.StateMachine.Action
+        
+        private(set) var estimatedStreamsPerConnection = 100
```

Comment:
probably a `static let` is better here.

Reply:
I removed it for now because we currently do not use it